text/0808-library-quality-scoring.md

```diff
@@ -0,0 +1,625 @@
+your own technical evaluation, security review, and testing before making adoption decisions.
+> [!NOTE]  
+> Each package is scored on their latest version. Scores are unlikely to change drasically between versions.
+
+#### CLI Usage
+
+```
+> cdk-construct-analyzer --help
+
+Usage: cdk-construct-analyzer <package> [options]
+
+Positionals:
+  package       NPM package name to analyze (Scored on the latest version)
+
+Options:
+ -v, --verbose  Show detailed signal information
+     --help     Show this help message
+     --version  Show version number
+```
+
+You can run it locally on any library published to npm by providing its package name:
+
+```
+> cdk-construct-analyzer cdk-ecr-deployment
+
+LIBRARY: @cdklabs/cdk-ecr-deployment
+VERSION: 0.0.421
+
+OVERALL SCORE: 81/100
+
+---
+
+SUBSCORES
+  Maintenance :            75/100
+  Quality     :            90/100
+  Popularity  :            88/100
+```
+
+##### Verbose
+
+Add `--verbose` for a detailed breakdown:
+
+```
+> cdk-construct-analyzer cdk-ecr-deployment --verbose
+
+LIBRARY: cdk-ecr-deployment
+VERSION: 0.0.421
+
+OVERALL SCORE: 81/100
+
+---
+
+SUBSCORES
+  Maintenance :            75/100
+  Quality     :            90/100
+  Popularity  :            88/100
+  
+---
+
+=== Maintenance ===                               SCORE  WEIGHT
+— Time to first response......................... ★★☆☆☆    3
+— Release Frequency ............................. ★★★★☆    3
+— Provenance Verification ....................... ★★★★★    3
+— Open issues / total issues .................... ★★★★★    3
+— Number of Contributors ........................ ★★★★☆    2
+
+=== Quality ===                                   SCORE  WEIGHT
+— Documentation Completeness .................... ★★★★★    3
+— Tests checklist (unit/snapshot) ............... ★★★☆☆    3
+— Author Track Record ........................... ★★★★★    3
+— Changelog includes feats/fixes ................ ★★★★★    3
+— Stable versioning ............................. ★★★★★    2
+— Multi-language Support ........................ ★★★★★    1
+
+=== Popularity ===                                SCORE  WEIGHT
+— Weekly Downloads .............................. ★★★★☆    3
+— Github stars .................................. ★★★★★    2
+— Contributors .................................. ★★★★☆    1
+```
+
+#### Programmatic Access
+
+You can also use the analyzer programmatically in your TypeScript applications by importing the `ConstructAnalyzer` class:
+
+```typescript
+import { ConstructAnalyzer } from '@cdklabs/cdk-construct-analyzer';
+
+const analyzer = new ConstructAnalyzer();
+
+// Analyze a package and get detailed results
+const result = await analyzer.analyzePackage('cdk-ecr-codedeploy');
+
+// Or analyze with custom weights
+const customWeights = {
+  'timeToFirstResponse': 15,
+  'commitFrequency': 10,
+  'releaseFrequency': 8,
+  // ... other signal weights that sum to 100
```

Comment:
does your library enforce this and throw an error? or will it automatically normalize to 100?